### PR TITLE
pd: add pd microservices section

### PR DESCRIPTION
**Pull request by rleungx**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [ ] master (the latest development version)
- [ ] v7.6 (TiDB 7.6 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.4 (TiDB 7.4 versions)
- [ ] v7.3 (TiDB 7.3 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: pd-microservices.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+- 可以避免 PD 集群压力过大而导致 TSO 分配的长尾或者抖动现象
+- 可以避免因为调度模块故障导致整个集群服务不可用的问题
+- 可以延缓 PD 自身单点瓶颈的问题
+
+此外，调度模块的变更可以单独更新 Scheduling 微服务，不需要再对 PD 进行重启，进而不会影响集群的整体服务。
+
+## 使用限制
+
+1. TSO 微服务目前不支持动态启停，开启或关闭需要重启 PD 集群。
+2. 只有 TiDB 通过服务发现直接连接 TSO 微服务，其他的组件是通过请求转发的方式，将请求通过 PD 转发到 TSO 微服务获取时间戳。
```
Comment (lhy1024) [inline pd-microservices.md:28]: 需要引入 API server 的概念么？

File: pd-microservices.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+微服务模式引入了如下命令行子命令：
+
+```
+pd-server services <mode>
+```
+
+`mode` 支持三种选项，具体如下：
+
+- `api`：若开启微服务，PD 自身将默认以 api mode 启动。启动后不再提供原本 TSO 分配的功能，需要在集群中同时部署 TSO 微服务。另外，原本的调度功能则根据是否部署 scheduling 微服务动态提供。
+- `tso`：若 mode 为 tso，即开启 TSO 微服务，提供时间戳分配的功能，需要 PD 以 api mode 启动。
```
Comment (lhy1024) [inline pd-microservices.md:41]: 需要指明无法在线地变更 tso 服务么？

File: pd-microservices.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+> **注意：**
+>
+> 无论是开启 TSO 还是 Scheduling 微服务，PD 都必须以 api mode 启动。
+> api mode 和 tso mode 相互依赖，需要集群中同时启用或停用。
+
+上述各种模式通常无需手动设置，由部署工具自动完成。
+
+此外，由于调度微服务是通过服务发现动态开启的，所以如果调度微服务发生宕机后，默认 PD 会继续提供调度的服务。但考虑到允许调度微服务和 PD 使用不同的版本，不同版本的调度逻辑可能会发生变化。所以也提供了开关，在这种情况下，禁止 PD 提供调度服务，防止调度逻辑出现跳变。可以通过设置 `pd-ctl config set enable-scheduling-fallback false` 进行关闭。该参数默认为 true。
+
+## 使用方法
```
Comment (lhy1024) [inline pd-microservices.md:53]: 需要提供一个 example 或者最佳实践？

**Review by lhy1024 — COMMENTED** [review]
LGTM

File: pd-microservices.md (hunk 3 above)
Comment (rleungx) [reply pd-microservices.md:53]: 等 operator 先合

File: pd-microservices.md (hunk 2 above)
Comment (rleungx) [reply pd-microservices.md:43]: 我补一下

File: TOC.md [hunk 4]
```diff
@@ -234,17 +234,17 @@
...
+    - [TiDB 内存调优](/configure-memory-usage.md)
+    - [TiKV 线程调优](/tune-tikv-thread-performance.md)
+    - [TiKV 内存调优](/tune-tikv-memory-performance.md)
+    - [TiKV Follower Read](/follower-read.md)
+    - [Region 性能调优](/tune-region-performance.md)
+    - [TiFlash 调优](/tiflash/tune-tiflash-performance.md)
+    - [下推计算结果缓存](/coprocessor-cache.md)
+    - 垃圾回收 (GC)
+      - [GC 机制简介](/garbage-collection-overview.md)
+      - [GC 配置](/garbage-collection-configuration.md)
```
Comment (qiancai) [inline TOC.md:247]:
```suggestion
      - [操作系统性能参数调优](/tune-operating-system.md)
      - [TiDB 内存调优](/configure-memory-usage.md)
      - [TiKV 线程调优](/tune-tikv-thread-performance.md)
      - [TiKV 内存调优](/tune-tikv-memory-performance.md)
      - [TiKV Follower Read](/follower-read.md)
      - [Region 性能调优](/tune-region-performance.md)
      - [TiFlash 调优](/tiflash/tune-tiflash-performance.md)
      - [下推计算结果缓存](/coprocessor-cache.md)
      - 垃圾回收 (GC)
        - [GC 机制简介](/garbage-collection-overview.md)
        - [GC 配置](/garbage-collection-configuration.md)
```

File: pd-microservices.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+每种微服务都以独立进程的方式部署，若指定的副本数量大于 1 时，提供主备的容灾模式。
+
+## 使用场景
+
+PD 微服务可以通过将重要模块，如 TSO 分配，调度等模块，与 PD 自身的路由功能进行解耦，让 PD 专注于元数据的路由服务。利用该特性：
+
+- 可以避免 PD 集群压力过大而导致 TSO 分配的长尾或者抖动现象
+- 可以避免因为调度模块故障导致整个集群服务不可用的问题
+- 可以延缓 PD 自身单点瓶颈的问题
```
Comment (qiancai) [inline pd-microservices.md:21]:
```suggestion
- PD 集群压力过大而导致 TSO 分配的长尾或者抖动现象
- 调度模块故障导致整个集群服务不可用的问题
- PD 自身单点瓶颈的问题
```

File: pd-microservices.md [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+从 v8.0.0 开始，PD 支持微服务模式。该模式可自定义将 PD 自身拆解成多个组件进行部署，具体包括
+
+- TSO 微服务：为整个集群提供单调递增的时间戳分配
+- Scheduling 微服务：为整个集群提供调度功能，包括但不限于负载均衡，热点，副本修复，副本放置等
+
+每种微服务都以独立进程的方式部署，若指定的副本数量大于 1 时，提供主备的容灾模式。
+
+## 使用场景
+
+PD 微服务可以通过将重要模块，如 TSO 分配，调度等模块，与 PD 自身的路由功能进行解耦，让 PD 专注于元数据的路由服务。利用该特性：
```
Comment (qiancai) [inline pd-microservices.md:17]:
```suggestion
PD 微服务通常用于解决 PD 出现性能瓶颈的问题，提高 PD 服务质量。利用该特性，你可以避免以下问题：
```

File: pd-microservices.md [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
+- 可以避免 PD 集群压力过大而导致 TSO 分配的长尾或者抖动现象
+- 可以避免因为调度模块故障导致整个集群服务不可用的问题
+- 可以延缓 PD 自身单点瓶颈的问题
+
+此外，调度模块的变更可以单独更新 Scheduling 微服务，不需要再对 PD 进行重启，进而不会影响集群的整体服务。
+
```
Comment (qiancai) [inline pd-microservices.md:24]: > **注意：**
>
>  如果瓶颈本身不在 PD，则无需开启，微服务本身会增加组件数量，提高运维成本。

File: pd-microservices.md [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+## 使用限制
+
+1. TSO 微服务目前不支持动态启停，开启或关闭需要重启 PD 集群。
+2. 只有 TiDB 通过服务发现直接连接 TSO 微服务，其他的组件是通过请求转发的方式，将请求通过 PD 转发到 TSO 微服务获取时间戳。
+3. 当前微服务与 dr-auto sync 特性不兼容。
```
Comment (qiancai) [inline pd-microservices.md:29]:
```suggestion
3. 当前微服务与 [同步部署模式 (DR Auto-Sync) ](/two-data-centers-in-one-city-deployment.md#简介) 特性不兼容。
```

File: pd-microservices.md (hunk 7 above)
Comment (qiancai) [inline pd-microservices.md:24]:
```suggestion

> **注意：**
>
>  如果瓶颈本身不在 PD，则无需开启，微服务本身会增加组件数量，提高运维成本。

```

File: pd-microservices.md [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+4. 与 TiDB 系统变量 `tidb_enable_tso_follower_proxy` 不兼容。
+
+## 使用方法
+
+目前仅支持通过 tidb-operator 进行部署。
```
Comment (qiancai) [inline pd-microservices.md:34]:
```suggestion
目前仅支持通过 TiDB Operator 进行部署。
```

File: pd-microservices.md [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+
+若开启微服务，PD 启动后不再提供原本 TSO 分配的功能，需要在集群中部署 TSO 微服务。另外，如果集群中部署了 scheduling 微服务，则由 scheduling 微服务提供调度功能，否则，由 PD 提供调度功能。
```
Comment (qiancai) [inline pd-microservices.md:36]:
```suggestion
开启微服务并重启 PD 后，PD 不再提供 TSO 分配功能，需要在集群中部署 TSO 微服务。另外，如果集群中部署了 scheduling 微服务，则由 scheduling 微服务提供调度功能，否则，由 PD 提供调度功能。
```

File: pd-microservices.md [hunk 11]
```diff
@@ -0,0 +1,58 @@
+---
+title: 使用 PD 微服务提高服务质量
+summary: 介绍如何开启 PD 微服务模式提高服务质量
```
Comment (qiancai) [inline pd-microservices.md:3]:
```suggestion
summary: 介绍如何开启 PD 微服务模式提高服务质量。
```

File: pd-microservices.md [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+
+## 使用方法
+
+目前仅支持通过 tidb-operator 进行部署。
+
+若开启微服务，PD 启动后不再提供原本 TSO 分配的功能，需要在集群中部署 TSO 微服务。另外，如果集群中部署了 scheduling 微服务，则由 scheduling 微服务提供调度功能，否则，由 PD 提供调度功能。
+
+> **注意：**
+>
+> 由于 scheduling 微服务支持动态切换，即如果 scheduling 微服务进程关闭后，PD 会继续提供调度的服务。所以如果 scheduling 微服务和 PD 使用不同的 binary 版本，为防止调度逻辑出现变化。可以通过设置 `pd-ctl config set enable-scheduling-fallback false` 禁止 scheduling 微服务进程关闭后 PD 提供调度服务。该参数默认为 true。
```
Comment (qiancai) [inline pd-microservices.md:40]:
```suggestion
> Scheduling 微服务支持动态切换，即如果 Scheduling 微服务进程关闭后，PD 默认会继续提供调度的服务。如果 scheduling 微服务和 PD 使用不同的 binary 版本，为防止调度逻辑出现变化，可以通过设置 `pd-ctl config set enable-scheduling-fallback false` 禁止 Scheduling 微服务进程关闭后 PD 提供调度服务。该参数默认为 `true`。
```

File: pd-microservices.md [hunk 13]
```diff
@@ -0,0 +1,58 @@
...
+
```
Comment (qiancai) [inline pd-microservices.md:41]: 问题，设置 `pd-ctl config set enable-scheduling-fallback false` 后，如果关闭Scheduling 微服务进程，PD 也不提供调度服务，那集群是不是就无法调度了

File: pd-microservices.md (hunk 9 above)
Comment (HuSharp) [reply pd-microservices.md:34]: 需要引入 playground 不？

Comment (rleungx) [reply pd-microservices.md:34]: 我觉得暂时不用吧

File: pd-microservices.md [hunk 14]
```diff
@@ -0,0 +1,57 @@
...
+> **注意：**
+>
+> 如果性能瓶颈本身不在 PD，则无需开启微服务。微服务本身会增加组件数量，提高运维成本。
+
+## 使用限制
+
+1. TSO 微服务目前不支持动态启停，开启或关闭需要重启 PD 集群。
+2. 只有 TiDB 通过服务发现直接连接 TSO 微服务，其他的组件是通过请求转发的方式，将请求通过 PD 转发到 TSO 微服务获取时间戳。
+3. 当前微服务与 [同步部署模式 (DR Auto-Sync) ](/two-data-centers-in-one-city-deployment.md#简介) 特性不兼容。
+4. 与 TiDB 系统变量 `tidb_enable_tso_follower_proxy` 不兼容。
```
Comment (qiancai) [inline pd-microservices.md:34]:
```suggestion
- TSO 微服务目前不支持动态启停，开启或关闭 TSO 微服务后需要重启 PD 集群。
- 只有 TiDB Server 可以通过服务发现直接连接 TSO 微服务，其它组件则需要将请求通过 PD 转发到 TSO 微服务获取时间戳。
- 当前微服务与[同步部署模式 (DR Auto-Sync)](/two-data-centers-in-one-city-deployment.md#简介) 特性不兼容。
- 与 TiDB 系统变量 [`tidb_enable_tso_follower_proxy`](/system-variables.md#tidb_enable_tso_follower_proxy-从-v530-版本开始引入) 不兼容。
```

File: pd-microservices.md [hunk 15]
```diff
@@ -0,0 +1,57 @@
...
+>
+> Scheduling 微服务支持动态切换，即如果 Scheduling 微服务进程关闭后，PD 默认会继续提供调度的服务。如果 Scheduling 微服务和 PD 使用不同的 binary 版本，为防止调度逻辑出现变化，可以通过设置 `pd-ctl config set enable-scheduling-fallback false` 禁止 Scheduling 微服务进程关闭后 PD 提供调度服务。该参数默认为 `true`。
+
+## 工具兼容性
+
+微服务不影响数据导入导出以及其他同步工具的正常使用。
+
+## 常见问题
+
+- 如何判断 PD 是否达到了性能瓶颈?
```
Comment (qiancai) [inline pd-microservices.md:52]:
```suggestion
- 如何判断 PD 是否达到了性能瓶颈?

```

File: pd-microservices.md [hunk 16]
```diff
@@ -0,0 +1,59 @@
...
+- 只有 TiDB 通过服务发现直接连接 TSO 微服务，其他的组件是通过请求转发的方式，将请求通过 PD 转发到 TSO 微服务获取时间戳。
+- 当前微服务与 [同步部署模式 (DR Auto-Sync) ](/two-data-centers-in-one-city-deployment.md#简介) 特性不兼容。
+- 与 TiDB 系统变量 `tidb_enable_tso_follower_proxy` 不兼容。
+- 由于静默 region 的关系，Scheduling 微服务在进行主备切换时，为避免冗余调度，集群或存在至多五分钟没有调度的现象。
+
+## 使用方法
+
+目前 PD 微服务仅支持通过 TiDB Operator 进行部署。
+
+开启微服务并重启 PD 后，PD 不再提供 TSO 分配功能，需要在集群中部署 TSO 微服务。此外，如果集群中部署了 scheduling 微服务，则由 scheduling 微服务提供调度功能，否则，由 PD 提供调度功能。
```
Comment (qiancai) [inline pd-microservices.md:41]:
```suggestion
开启微服务并重启 PD 后，PD 不再提供 TSO 分配功能，需要在集群中部署 TSO 微服务。此外，如果集群中部署了 Scheduling 微服务，则由 Scheduling 微服务提供调度功能，否则，由 PD 提供调度功能。
```

File: pd-microservices.md [hunk 17]
```diff
@@ -0,0 +1,59 @@
...
+---
+
+# 使用 PD 微服务模式
+
+从 v8.0.0 开始，PD 支持微服务模式。该模式可将 PD 的时间戳分配和集群调度功能拆分为以下微服务单独部署，从而与 PD 的路由功能解耦，让 PD 专注于元数据的路由服务。
+
+- TSO 微服务：为整个集群提供单调递增的时间戳分配。
+- Scheduling 微服务：为整个集群提供调度功能，包括但不限于负载均衡、热点处理、副本修复、副本放置等。
+
+每种微服务都以独立进程的方式部署，当相应服务设置的副本数量大于 1 时，提供主备的容灾模式。
```
Comment (qiancai) [inline pd-microservices.md:13]:
```suggestion
每种微服务都以独立进程的方式部署。当相应微服务设置的副本数大于 1 时，该微服务支持主备的容灾模式。
```

File: pd-microservices.md [hunk 18]
```diff
@@ -0,0 +1,59 @@
...
+此外，调度模块的变更可以单独更新 Scheduling 微服务，不需要再对 PD 进行重启，进而不会影响集群的整体服务。
+
+> **注意：**
+>
+> 如果性能瓶颈本身不在 PD，则无需开启微服务。微服务本身会增加组件数量，提高运维成本。
+
+## 使用限制
+
+- TSO 微服务目前不支持动态启停，开启或关闭需要重启 PD 集群。
+- 只有 TiDB 通过服务发现直接连接 TSO 微服务，其他的组件是通过请求转发的方式，将请求通过 PD 转发到 TSO 微服务获取时间戳。
```
Comment (qiancai) [inline pd-microservices.md:32]:
```suggestion
- 只有 TiDB 组件支持通过服务发现直接连接 TSO 微服务，其他的组件是通过请求转发的方式，将请求通过 PD 转发到 TSO 微服务获取时间戳。
```

File: pd-microservices.md [hunk 19]
```diff
@@ -0,0 +1,59 @@
...
+- 当前微服务与 [同步部署模式 (DR Auto-Sync) ](/two-data-centers-in-one-city-deployment.md#简介) 特性不兼容。
+- 与 TiDB 系统变量 `tidb_enable_tso_follower_proxy` 不兼容。
+- 由于静默 region 的关系，Scheduling 微服务在进行主备切换时，为避免冗余调度，集群或存在至多五分钟没有调度的现象。
```
Comment (qiancai) [inline pd-microservices.md:35]:
```suggestion
- 由于集群中可能存在静默 Region，Scheduling 微服务在进行主备切换时，为避免冗余调度，集群可能存在至多五分钟没有调度的现象。
```

File: pd-microservices.md [hunk 20]
```diff
@@ -0,0 +1,59 @@
...
+
+## 使用方法
+
+目前 PD 微服务仅支持通过 TiDB Operator 进行部署。
+
+开启微服务并重启 PD 后，PD 不再提供 TSO 分配功能，需要在集群中部署 TSO 微服务。此外，如果集群中部署了 scheduling 微服务，则由 scheduling 微服务提供调度功能，否则，由 PD 提供调度功能。
+
+> **注意：**
+>
+> Scheduling 微服务支持动态切换，即如果 Scheduling 微服务进程关闭后，PD 默认会继续提供调度的服务。如果 Scheduling 微服务和 PD 使用不同的 binary 版本，为防止调度逻辑出现变化，可以通过设置 `pd-ctl config set enable-scheduling-fallback false` 禁止 Scheduling 微服务进程关闭后 PD 提供调度服务, 但是可能在 Scheduling 微服务进程重新启动前，集群将无法提供调度服务。该参数默认为 `true`。
```
Comment (qiancai) [inline pd-microservices.md:45]:
```suggestion
> - Scheduling 微服务动态切换功能默认开启（`enable-scheduling-fallback` 默认为 `true`)。如果 Scheduling 微服务进程关闭后，PD 默认会继续为集群提供调度服务。
> - 如果 Scheduling 微服务和 PD 使用的 binary 版本不同，为防止调度逻辑出现变化，可以通过设置 `pd-ctl config set enable-scheduling-fallback false` 禁止 PD 在 Scheduling 微服务进程关闭后提供调度服务。此时，在 Scheduling 微服务进程重新启动前，集群将无法提供调度服务。
```

File: pd-microservices.md [hunk 21]
```diff
@@ -0,0 +1,59 @@
...
+
+- TSO 微服务目前不支持动态启停，开启或关闭需要重启 PD 集群。
+- 只有 TiDB 通过服务发现直接连接 TSO 微服务，其他的组件是通过请求转发的方式，将请求通过 PD 转发到 TSO 微服务获取时间戳。
+- 当前微服务与 [同步部署模式 (DR Auto-Sync) ](/two-data-centers-in-one-city-deployment.md#简介) 特性不兼容。
+- 与 TiDB 系统变量 `tidb_enable_tso_follower_proxy` 不兼容。
+- 由于静默 region 的关系，Scheduling 微服务在进行主备切换时，为避免冗余调度，集群或存在至多五分钟没有调度的现象。
+
+## 使用方法
+
+目前 PD 微服务仅支持通过 TiDB Operator 进行部署。
```
Comment (qiancai) [inline pd-microservices.md:39]:
```suggestion
目前 PD 微服务仅支持通过 TiDB Operator 进行部署。部署方法请参考 [TiDB Operator 文档](https://docs.pingcap.com/zh/tidb-in-kubernetes/dev/configure-a-tidb-cluster#部署-pd-微服务)。
```

File: pd-microservices.md [hunk 22]
```diff
@@ -0,0 +1,59 @@
...
+
+## 常见问题
+
+- 如何判断 PD 是否达到了性能瓶颈?
+
+  在集群自身状态正常的前提下，可以查看 Grafana PD 面板中的监控指标。如果 `TiDB - PD server TSO handle time` 指标出现明显延迟上涨或 `Heartbeat - TiKV side heartbeat statistics` 指标出现大量 pending，说明 PD 达到了性能瓶颈。
+
+## 另请参阅
+
+TODO
```
Comment (qiancai) [inline pd-microservices.md:59]:
```suggestion
- [部署 PD 微服务](https://docs.pingcap.com/zh/tidb-in-kubernetes/dev/configure-a-tidb-cluster#部署-pd-微服务)
- [配置 PD 微服务](https://docs.pingcap.com/zh/tidb-in-kubernetes/dev/configure-a-tidb-cluster#配置-pd-微服务)
- [修改 PD 微服务](https://docs.pingcap.com/zh/tidb-in-kubernetes/dev/modify-tidb-configuration#配置-pd-微服务)
- [扩缩容 PD 微服务组件](https://docs.pingcap.com/zh/tidb-in-kubernetes/dev/scale-a-tidb-cluster#扩缩容-pd-微服务组件)
```

**Comment by qiancai** [comment]
Close this PR as its content has been moved to https://github.com/pingcap/docs-cn/pull/15975

**Pull request closed without merging**